frame/contracts/src/benchmarking/mod.rs

```diff
@@ -893,15 +892,75 @@ benchmarks! {
 			}],
 			call_body: Some(body::repeated(r * API_BENCHMARK_BATCH_SIZE, &[
 				Instruction::I32Const(0), // value_ptr
-				Instruction::I32Const(max_bytes as i32), // value_len
+				Instruction::I32Const(0), // value_len
 				Instruction::Call(0),
 				Instruction::Drop,
 			])),
 			.. Default::default()
 		});
 		let instance = Contract::<T>::new(code, vec![])?;
-		let origin = RawOrigin::Signed(instance.caller.clone());
-	}: call(origin, instance.addr, 0u32.into(), Weight::MAX, None, vec![])
+	}: {
+		<Contracts<T>>::bare_call(
+			instance.caller,
+			instance.account_id,
+			0u32.into(),
+			Weight::MAX,
+			None,
+			vec![],
+			true,
+			Determinism::Deterministic,
+		)
+		.result?;
+	}
+
+	seal_debug_message_per_kb {
+		// Vary size of input in kilobytes up to maximum allowed contract memory.
+		let i in 0 .. (T::Schedule::get().limits.memory_pages * 64);
```

Comment:
This should only run up to the amount you actually initialised: `T::MaxCodeLen::get() / 1024 - 1`

Reply:
Why? All uninitialized bytes get `0` value, which is a proper UTF-8 code (`NUL`), so we can actually test versus longer inputs having it the way it is now. 

We just have to initialize a smaller part because of the contracts code length restriction:  the initializers (data segments) for the whole contract memory would just make the benchmark-generated contract size exceed the `MaxCodeLen` limit. That's being said, the debug message input could really exceed the size we initialize here in the benchmark.

Reply:
You are right. I was just afraid that it stops parsing at the first zero byte because of null termination. But Rust ist UTF-8 compliant and should allow nulls within strings.

Reply:
True, the host function just [cuts the passed message](https://github.com/paritytech/substrate/blob/ag-debug-buffer/frame/contracts/src/wasm/runtime.rs#L2385) to the `MaxDebugBufferLen` size. So if we had that bound less than maximum allowed contract memory (which is not the case currently), the benchmark result would be distorted. I corrected the benchmark to vary its input up to whichever is less among of the two bounds.